src/Nethermind/Nethermind.Blockchain/LogTraceDumper.cs

```diff
@@ -11,25 +11,33 @@
 using Nethermind.Evm.Tracing.ParityStyle;
 using Nethermind.Logging;
 using Nethermind.Serialization.Json;
+using Nethermind.Serialization.Rlp;
 using Newtonsoft.Json;
 
 namespace Nethermind.Blockchain;
 
 public static class BlockTraceDumper
 {
-    public static List<JsonConverter> Converters { get; } = new List<JsonConverter>();
+    public static List<JsonConverter> Converters { get; } = new();
+
+    public static void LogDiagnosticRlp(
+        Block block,
+        ILogger logger)
+    {
+        Keccak blockHash = block.Hash;
+        string fileName = $"block_{blockHash}.txt";
```

Comment:
The `.txt` extension is misleading as the file is binary. Let's either remove it or change to `.rlp`:

```suggestion
        string fileName = $"block_{blockHash}.rlp";
```